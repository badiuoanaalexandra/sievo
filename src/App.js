import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux'
import { requestProjectsData} from './actions/projectsApiActions'

import ProjectTable from './components/table/index';
import Filter from './components/filter/index';

class App extends React.Component {
  componentWillMount() {
    this.props.requestProjectsData();
  }
  render() {
    return (
      <div className="App">
          <Filter />
          <ProjectTable />
      </div>
    );
  }
}

const mapStateToProps = state => ({ data:state.dataReducer })

const mapDispatchToProps = dispatch => bindActionCreators({requestProjectsData}, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(App);
