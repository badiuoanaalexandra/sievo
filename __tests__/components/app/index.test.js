import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import App from '../../../src/components/app/index';

const mockStore = configureStore();

const initialState = {
  projectsReducer: [
      {"project":2, "description":"test", "start date": "2013-12-28T00:00:00+00:00", "savings amount":3722.41684, "currency":"NULL"},
      {"project":1, "description":"test 2", "start date": "2014-01-28T00:00:00+00:00", "savings amount":372.32, "currency":"EUR"}
  ],
  displayProjectsReducer:{changed:false, description:"", sortingOrder:"", changedProjects:null},
  uiReducer:1024
};

const store = mockStore(initialState);

describe('<App />', () => {
  test("renders", () => {
    const wrapper = shallow(<App store={store} />);
    const component = wrapper.dive();
    expect(toJson(component)).toMatchSnapshot();
  });
});
