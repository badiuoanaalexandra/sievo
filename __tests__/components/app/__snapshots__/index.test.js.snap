// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
<div
  className="app"
>
  <div
    className="background-image-wrapper"
  >
    <img
      src="background-desktop.jpg"
    />
  </div>
  <div
    className="background-image-content"
  >
    <Connect(Filter) />
    <Connect(ProjectTable) />
  </div>
</div>
`;
